### PR TITLE
[WFLY-4563] Update log categories and messages where needed

### PR DESCRIPTION
**Pull request by jamezp**
Several updates to log messages and log categories. Just some general clean up.

The EJB change could use a good look as some new categories were added to help an end user be able to configure loggers for better debugging.

I also tried to replace `\n` line feeds to `System.lineSeparator()` where I could find them for better cross platform support.

Finally some places I removed the `isLevelEnabled()` and some places I guessed at adding where I felt the value replacement arguments may be expensive to retrieve.

### CODE REVIEW

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/EjbJndiBindingsDeploymentUnitProcessor.java [hunk 1]
```diff
@@ -128,7 +122,8 @@ private void setupJNDIBindings(EJBComponentDescription sessionBean, DeploymentUn
 
         // the base ServiceName which will be used to create the ServiceName(s) for each of the view bindings
         final StringBuilder jndiBindingsLogMessage = new StringBuilder();
-        jndiBindingsLogMessage.append("JNDI bindings for session bean named " + sessionBean.getEJBName() + " in deployment unit " + deploymentUnit + " are as follows:\n\n");
+        jndiBindingsLogMessage.append("JNDI bindings for session bean named " + sessionBean.getEJBName() + " in deployment unit " + deploymentUnit + " are as follows:");
```
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/EjbJndiBindingsDeploymentUnitProcessor.java]: I opened https://issues.jboss.org/browse/WFLY-4612 to get this fixed.

File: jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java [hunk 2]
```diff
@@ -62,8 +62,6 @@
     public static final String PESSIMISTIC_WRITE_LOCK_MODE = "pessimistic_write";
     public static final String NONE_LOCK_MODE = "none";
 
-    private final transient boolean isTraceEnabled = ROOT_LOGGER.isTraceEnabled();
```
Comment (bstansberry) [inline jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java]: @scottmarlow was this done this way for perf reasons?

File: jpa/src/main/java/org/jboss/as/jpa/container/ExtendedEntityManager.java [hunk 3]
```diff
@@ -89,8 +89,6 @@
      */
     private final ExtendedEntityManagerKey ID = ExtendedEntityManagerKey.extendedEntityManagerID();
 
-    private final transient boolean isTraceEnabled = ROOT_LOGGER.isTraceEnabled();
```
Comment (bstansberry) [inline jpa/src/main/java/org/jboss/as/jpa/container/ExtendedEntityManager.java]: @scottmarlow same here

File: naming/src/main/java/org/jboss/as/naming/ServiceBasedNamingStore.java [hunk 4]
```diff
@@ -154,9 +154,7 @@ public Object run() {
             n.initCause(e);
             throw n;
         } catch (Throwable t) {
-            NamingException n = NamingLogger.ROOT_LOGGER.lookupError(name);
-            n.initCause(t);
-            throw n;
+            throw NamingLogger.ROOT_LOGGER.lookupError(t, name);
```
Comment (bstansberry) [inline naming/src/main/java/org/jboss/as/naming/ServiceBasedNamingStore.java]: The logging tooling handles this automatically? If so, nice!

Comment (jamezp) [inline naming/src/main/java/org/jboss/as/naming/ServiceBasedNamingStore.java]: Yes it does :)

File: jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java (hunk 2 above)
Comment (jamezp) [inline jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java]: If it's done like this the trace logging for this class won't work until the server is restarted or at least until a new instance is created.

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/EjbJndiBindingsDeploymentUnitProcessor.java (hunk 1 above)
Comment (jamezp) [inline ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/EjbJndiBindingsDeploymentUnitProcessor.java]: Should I remove this? In fact I should have cleaned up the string concatenation too.

File: jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java (hunk 2 above)
Comment (bstansberry) [inline jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java]: It's an instance variable, and I believe, but could be wrong, that these are per-request objects. If so, then the current code is a reasonable perf optimization by removing possibly numerous calls to isTraceEnabled(). The only downside is the inability to react to a logging config change in the middle of a request.

Comment (scottmarlow) [inline jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java]: Yes, its intended as a performance optimization, since the AbstractEntityManager methods are called very often by applications.  The scope of the objects varies (could be per transaction, per component invocation or stateful bean lifetime (xpc)).

File: jpa/src/main/java/org/jboss/as/jpa/container/ExtendedEntityManager.java (hunk 3 above)
Comment (scottmarlow) [inline jpa/src/main/java/org/jboss/as/jpa/container/ExtendedEntityManager.java]: The scope of the object here is more likely to be per referencing stateful bean(s).  I haven't gotten any complaints about this.  If TRACE logging is enabled after starting the app server (if that is possible?), the application would have to either get lucky (create new stateful beans) or redeploy to cause new instances to be created.

File: jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java (hunk 2 above)
Comment (bstansberry) [inline jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java]: Thanks, Scott. IMO the change in this particular commit is out of scope for this JIRA.

Comment (jamezp) [inline jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java]: Okay so I should revert this and add the instance variable back then, correct? It's fine it just won't support enabling trace logging at runtime.

Comment (bstansberry) [inline jpa/src/main/java/org/jboss/as/jpa/container/AbstractEntityManager.java]: Right; the commit that does this part can just rebased out, or the change reverted in some other fashion.

Trace logging config changes will still be picked up by newly instantiated objects, and in most cases the lifespan of these objects is short. Even in the SFSB with XPC case it will probably be fine in practice, as new SFSBs will be instantiated frequently enough to get useful trace logging.

**Comment by jamezp** [comment]
Reverted the commit for the `isTraceEnabled` in JPA and rebased on the current master.

File: ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/EjbJndiBindingsDeploymentUnitProcessor.java (hunk 1 above)
Comment (bstansberry) [inline ejb3/src/main/java/org/jboss/as/ejb3/deployment/processors/EjbJndiBindingsDeploymentUnitProcessor.java]: Don't remove it, as it's important logging. You don't even need to deal with it as part of this PR; I added the comment mostly as an FYI.

**Comment by n1hility** [comment]
needs a rebase
